app/main.py

```diff
@@ -1,214 +1,26 @@
-KNIGHTS = {
-    "lancelot": {
-        "name": "Lancelot",
-        "power": 35,
-        "hp": 100,
-        "armour": [],
-        "weapon": {
-            "name": "Metal Sword",
-            "power": 50,
-        },
-        "potion": None,
-    },
-    "arthur": {
-        "name": "Arthur",
-        "power": 45,
-        "hp": 75,
-        "armour": [
-            {
-                "part": "helmet",
-                "protection": 15,
-            },
-            {
-                "part": "breastplate",
-                "protection": 20,
-            },
-            {
-                "part": "boots",
-                "protection": 10,
-            }
-        ],
-        "weapon": {
-            "name": "Two-handed Sword",
-            "power": 55,
-        },
-        "potion": None,
-    },
-    "mordred": {
-        "name": "Mordred",
-        "power": 30,
-        "hp": 90,
-        "armour": [
-            {
-                "part": "breastplate",
-                "protection": 15,
-            },
-            {
-                "part": "boots",
-                "protection": 10,
-            }
-        ],
-        "weapon": {
-            "name": "Poisoned Sword",
-            "power": 60,
-        },
-        "potion": {
-            "name": "Berserk",
-            "effect": {
-                "power": +15,
-                "hp": -5,
-                "protection": +10,
-            }
-        }
-    },
-    "red_knight": {
-        "name": "Red Knight",
-        "power": 40,
-        "hp": 70,
-        "armour": [
-            {
-                "part": "breastplate",
-                "protection": 25,
-            }
-        ],
-        "weapon": {
-            "name": "Sword",
-            "power": 45
-        },
-        "potion": {
-            "name": "Blessing",
-            "effect": {
-                "hp": +10,
-                "power": +5,
-            }
-        }
-    }
-}
+from app.battle_preparation.initial_data import knights
+from app.knights.knight import knight_create
+from app.battle_preparation.battle_preparation import knights_power_calculation
 
 
-def battle(knightsConfig):
-    # BATTLE PREPARATIONS:
+def battle(knights_in_dictionary: dict) -> dict:
+    game_knights = [knight_create(knights_in_dictionary, knight)
+                    for knight in knights_in_dictionary]
+    ready_knights = [knights_power_calculation(knight)
+                     for knight in game_knights]
 
-    # lancelot
-    lancelot = knightsConfig["lancelot"]
+    ready_knights[0].hp -= ready_knights[2].power - ready_knights[0].protection
+    ready_knights[1].hp -= ready_knights[3].power - ready_knights[1].protection
+    ready_knights[2].hp -= ready_knights[0].power - ready_knights[2].protection
+    ready_knights[3].hp -= ready_knights[1].power - ready_knights[3].protection
```

Comment:
You could make a separate function that performs these calculations over two knights.